src/eva/data/mon_data_space.py

```diff
@@ -77,43 +92,70 @@ def execute(self, dataset_config, data_collections, timing):
         channo = chans_dict["chan_nums"] if chans_dict is not None else None
         x_range, y_range, z_range = self.get_dim_ranges(coords, dims, channo)
 
-        # Filenames to be read into this collection
-        # -----------------------------------------
-        filenames = get(dataset_config, self.logger, 'filenames')
-        ds_list = []
-
         # Get missing value threshold
         # ---------------------------
         threshold = float(get(dataset_config, self.logger, 'missing_value_threshold', 1.0e30))
 
+        ds_list = []
+        file_ctr = 1
         for filename in filenames:
 
-            # read data file
-            darr, cycle_tm = self.read_ieee(filename, coords, dims, ndims_used,
-                                            dims_arr, nvars, vars)
+            lat = []
+            lon = []
+            if stn_data:
+
+                # Read station data file.  Note that there is no guarantee the number
+                # of variables is the same for different station data files.  The file_ctr
+                # is used to tack on a number to distinguish the coordinates/dimensions
+                # and corresponding variables for file1, file2, etc.
+                darr, cycle_tm, dims, lat, lon = self.read_stn_ieee(filename, coords, dims,
+                                                                    ndims_used, dims_arr, nvars,
+                                                                    stn_vars)
+                y_range = np.arange(1, dims['ydef']+1)
+                coords['ydef'] = "Nobs" + str(file_ctr)
```

Comment:
Does this string need a space in it? `"Nobs " + str(file_ctr)`?

Reply:
I guess it could, but I don't think it's necessary.  The problem is that the number of obs is file dependent.  So I need a means to associate the variables in a given file with the correct coordinates and dimensions when loading more than one (the ozn horiz plots load 4 files).  This solution gives me variables obs1, ges1, etc associated with dimension nobs1 and coordinate Nobs1, and then a different set with the next file.  Horiz plots typically load 1-4 files.

Alternatively I could figure out the greatest number of obs in the fileset and pad the shorter ones out with NaN values so the dimensions could be shared by all variables. That would be a bit more of more effort in the data load but I'll be glad to go that way if you'd prefer.  Now that I describe it I think I like it better than what I've got now.

Reply:
Ok I understand better what the string concatenation is doing. I'll defer to you. If you are happy with what you have, so am I but I also will support your suggestion of shared dimensions (it would probably be cleaner)

Reply:
Let me give that a try.  If it doesn't take to long it's probably worth it.  I'll switch this PR to draft.

Reply:
Revising that was the right answer.  I learned about xarray.align() and the awesome power that wields.  One call syncs the dimensions of all datasets and fills missing values with NaN.  It sure is easy when you know how. :)